contracts/modules/governance/external/Proxy.sol

```diff
@@ -1,28 +1,44 @@
 // SPDX-License-Identifier: GPL-3.0-only
 
-pragma solidity ^0.5.0;
+pragma solidity ^0.8.16;
 
 /**
  * @title Proxy
  * @dev Gives the possibility to delegate any call to a foreign implementation.
  */
-contract Proxy {
+abstract contract Proxy {
 
   /**
   * @dev Fallback function allowing to perform a delegatecall to the given implementation.
   * This function will return whatever the implementation call returns
   */
-  // solhint-disable-next-line no-complex-fallback
-  function() external payable {
+  fallback()  external payable {
     address _impl = implementation();
     require(_impl != address(0));
 
     // solhint-disable-next-line no-inline-assembly
     assembly {
       let ptr := mload(0x40)
-      calldatacopy(ptr, 0, calldatasize)
-      let result := delegatecall(gas, _impl, ptr, calldatasize, 0, 0)
-      let size := returndatasize
+      calldatacopy(ptr, 0, calldatasize())
+      let result := delegatecall(gas(), _impl, ptr, calldatasize(), 0, 0)
+      let size := returndatasize()
+      returndatacopy(ptr, 0, size)
+
+      switch result
+      case 0 {revert(ptr, size)}
+      default {return (ptr, size)}
+    }
+  }
+
+  receive() external payable {
```

Comment:
The code looks functionally correct. I compared this Proxy.sol from the OZ v4 we have installed and the only difference is the memory offset at which it starts, 0x40 but we can safely ignore it. The only thing I'd do here is avoid the duplicated code. Please take a look at
`node_modules/@openzeppelin/contracts-v4/proxy/Proxy.sol` and implement a `_delegate` function that would contain the assembly. The `fallback` and `receive` functions should call `_delegate`. Ignore the v4 hooks.